evm_arithmetization/src/witness/traces.rs

```diff
@@ -116,42 +116,60 @@ impl<T: Copy> Traces<T> {
         &self.memory_ops[checkpoint.memory_len..]
     }
 
-    pub(crate) fn push_cpu(&mut self, val: CpuColumnsView<T>) {
-        self.cpu.push(val);
+    pub(crate) fn push_cpu(&mut self, is_generation: bool, val: CpuColumnsView<T>) {
+        if is_generation {
+            self.cpu.push(val);
+        }
```

Comment:
I think a way to alleviate these kind of bad-looking methods is to have them defined one layer above, as `Traces` are a field of `GenerationState`.

Having them defined as part of the `State` trait would allow getting rid of the `is_generation` method and booleans floating around everywhere, and these would be a no-op for the Interpreter state.